packages/pending_operations/lib/src/pending_operations.dart

```diff
@@ -9,7 +9,7 @@ class PendingOperations implements PendingOperationsReader {
 
   int _pendingCalls = 0;
 
-  Future<void> _knownCallsHandled = Future<void>.value();
+  List<Future<void>> _knownCallsHandled = [];
```

Comment:
Previously this actually got stuck for me. Now we are only tracking the futures and then use `Future.wait` only in `awaitEndOfKnownCalls `

Reply:
Hmm, but the evaluation shouldn't change, `awaitEndOfKnownCalls` still has to be called in `tester.runAsync`. I just thought its most efficient to only keep track of the last future instead of all (not sure what else is kept alive by this.)

Will look into this in the overall scope again.

Reply:
As discussed: All of the futures are kept alive. It's just different in how we store them. Now we store in a flat list as before we had a nested list of `n` with always two items inside so we potentially had the overhead of `n * Futures.wait` in my opinion.

Reply:
Guess it's fine for now as `awaitEndOfKnownCalls` is only used for testing; but I still want to look into it in the future, there should be a cleaner way that doesn't grow memory.